### PR TITLE
Update ppo_trainer.py

### PR DESCRIPTION
**Pull request by skaulintel**
This changes helpes speed up performance of the backwards loss calculation with hpu graphs.

### CODE REVIEW

**Comment by skaulintel** [comment]
Here is the performance with and without the change

**baseline** | **with change**
1.940009 | 2.350880623
5.784701 | 0.04837203
4.520658 | 0.065311909
0.121894 | 0.064571142
0.120589 | 0.064719439
0.120764 | 0.064688444
0.100668 | 0.065456629
0.120939 | 0.064553976
0.103489 | 0.047488451
0.102377 | 0.064978123
0.099564 | 0.065481186
0.099492 | 0.065706968
0.099102 | 0.06585598
0.099199 | 0.065846682
0.098894 | 0.065851927
0.099187 | 0.065863848
0.110397 | 0.04795146
0.108176 | 0.064869642
0.10726 | 0.065325499
0.106228 | 0.065702677
0.106589 | 0.065702438
0.106565 | 0.065774202
0.105649 | 0.065782547
0.10621 | 0.065749645
0.099913 | 0.044877529
0.106869 | 0.065789223
0.106403 | 0.065751553
0.105779 | 0.06581831

File: optimum/habana/trl/trainer/ppo_trainer.py [hunk 1]
```diff
@@ -558,7 +616,8 @@ def step(
 
             model = self.accelerator.unwrap_model(self.model)
             if not hasattr(model, "wrap_train_in_graph"):
-                ht.hpu.ModuleCacher()(model=model, inplace=True)
+                model.eval()
```
Comment (sywangyi) [inline optimum/habana/trl/trainer/ppo_trainer.py:619]: do we need model.eval here? it's set in https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/trl/trainer/ppo_trainer.py#L745

File: optimum/habana/trl/trainer/ppo_trainer.py [hunk 2]
```diff
@@ -820,10 +913,24 @@ def train_minibatch(
...
+        if _recorded_graph is None:
+            _recorded_graph = ht.hpu.HPUGraph()
+            s = ht.hpu.default_stream()
+
+            with ht.hpu.stream(s):
+                _recorded_graph.capture_begin()
+                self.accelerator.backward(loss)
+                _recorded_graph.capture_end()
+        else:
+            ht.core.mark_step()
```
Comment (amitchawla1) [inline optimum/habana/trl/trainer/ppo_trainer.py:927]: Replay will have internal mark_step, this mark_step is not required.

Comment (skaulintel) [reply optimum/habana/trl/trainer/ppo_trainer.py:927]: removed

**Review by sywangyi — APPROVED** [review]

File: optimum/habana/trl/trainer/ppo_trainer.py [hunk 3]
```diff
@@ -820,7 +833,18 @@ def train_minibatch(
...
+        if _recorded_graph is None:
+            _recorded_graph = ht.hpu.HPUGraph()
+            s = ht.hpu.default_stream()
+
+            with ht.hpu.stream(s):
+                _recorded_graph.capture_begin()
+                self.accelerator.backward(loss)
+                _recorded_graph.capture_end()
+        else:
+            _recorded_graph.replay()
```
Comment (regisss) [inline optimum/habana/trl/trainer/ppo_trainer.py:847]: Out of curiosity, would that make sense to have the same logic in the `GaudiTrainer`?

**Review by regisss — APPROVED** [review]
LGTM!
